src/common/src/types/num256.rs

```diff
@@ -64,7 +77,34 @@ macro_rules! impl_common_for_num256 {
             }
         }
 
+        impl<'de> Deserialize<'de> for $scalar {
+            fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+            where
+                D: serde::Deserializer<'de>,
+            {
+                <$inner>::deserialize(deserializer).map(Into::into)
+            }
+        }
```

Comment:
We cannot use `Serialize`/`Deserialize` on `ethnum::I256` directly, which is not memcomparable:
```
values (16::int256), (2), (1) order by 1;
```

https://docs.rs/ethnum/1.3.2/src/ethnum/serde.rs.html#47

It is encoding them as variable length hex string `0x10`, `0x2`, `0x1` so that 1 < 16 < 2. We should use our own implementation. Beware of negative values as well.